cpp/src/arrow/array/builder_nested.h

```diff
@@ -672,6 +672,7 @@ class ARROW_EXPORT FixedSizeListBuilder : public ArrayBuilder {
   /// using the child array builder.
   Status Append();
 
+  void UnsafeAppend();
```

Comment:
1. Can you please add docstrings for these new methods?
2. Ideally, the definition of the methods should be inline here, so that they can be inlined in the caller.